### PR TITLE
fix(vscode): Do not use Microsoft marketplace for French extension

### LINKED ISSUE #54: Do not use Msft Marketplace for i18n of vscode
Relating to work done in #47 for non-Microsoft implementations of VS Code you are not supposed to access Microsoft's marketplace. 
https://github.com/cdr/code-server/blob/master/doc/FAQ.md#differences-compared-to-vs-code
The extension I used was taken from the Microsoft Marketplace via a wget and then placed used by code server.

### PR DESCRIPTION
**Pull request by Jose-Matsuda**
-not supposed to use microsoft marketplace for cdr
-instead build from github repo
 -https://github.com/microsoft/vscode-loc/tree/release/1.50
-steps to build found in the wiki of said repo above
-removes downloaded files to keep image small

Example of run [here](https://streamable.com/vaklkg)

### CODE REVIEW

**Comment by Jose-Matsuda** [comment]
**BEFORE REVIEW / MAKE CHANGES**
USE GIT CLONE TO GET THE REPO instead of **wget**'ing the zip file

**resolved**

File: base/Dockerfile [hunk 1]
```diff
@@ -579,13 +578,21 @@ RUN \
     bsdtar -xf gitlens-$VS_GITLENS_VERSION.vsix extension && \
     rm gitlens-$VS_GITLENS_VERSION.vsix && \
     mv extension $HOME/.vscode/extensions/eamodio.gitlens-$VS_GITLENS_VERSION && \
-    #Install French Language extension: https://marketplace.visualstudio.com/items?itemName=MS-CEINTL.vscode-language-pack-fr
-    VS_FRENCH_VERSION="1.44.0" && \
-    wget --quiet --no-check-certificate --content-disposition https://marketplace.visualstudio.com/_apis/public/gallery/publishers/MS-CEINTL/vsextensions/vscode-language-pack-fr/$VS_FRENCH_VERSION/vspackage &&\
-    echo "${SHA256fr} MS-CEINTL.vscode-language-pack-fr-${VS_FRENCH_VERSION}.vsix" | sha256sum -c - && \
-    bsdtar -xf MS-CEINTL.vscode-language-pack-fr-$VS_FRENCH_VERSION.vsix extension && \
-    rm MS-CEINTL.vscode-language-pack-fr-$VS_FRENCH_VERSION.vsix && \
+    #Install French Language extension via building and copying https://github.com/microsoft/vscode-loc/tree/release/1.50
```
Comment (wg102) [inline base/Dockerfile]: - Why use version 1.50 and not the newer https://github.com/microsoft/vscode-loc/tree/release/1.52 ?
- Start the comment line with space

File: base/Dockerfile [hunk 2]
```diff
@@ -579,13 +578,21 @@ RUN \
...
+    VS_LOCALE_REPO_VERSION="1.50" && \
+    git clone -b release/$VS_LOCALE_REPO_VERSION https://github.com/microsoft/vscode-loc.git &&\
+    cd vscode-loc && \
+    npm install -g vsce && \
+    cd i18n/vscode-language-pack-fr && \
+    vsce package && \
+    bsdtar -xf vscode-language-pack-fr-$VS_FRENCH_VERSION.vsix extension && \
     mv extension $HOME/.vscode/extensions/ms-ceintl.vscode-language-pack-fr-$VS_FRENCH_VERSION && \
+    cd ../../../ && \
+    # -fr as the clone becomes a write-protected regular file
```
Comment (wg102) [inline base/Dockerfile]: Clearer comment would be useful

File: base/Dockerfile [hunk 3]
```diff
@@ -579,13 +578,21 @@ RUN \
...
-    #Install French Language extension: https://marketplace.visualstudio.com/items?itemName=MS-CEINTL.vscode-language-pack-fr
-    VS_FRENCH_VERSION="1.44.0" && \
-    wget --quiet --no-check-certificate --content-disposition https://marketplace.visualstudio.com/_apis/public/gallery/publishers/MS-CEINTL/vsextensions/vscode-language-pack-fr/$VS_FRENCH_VERSION/vspackage &&\
-    echo "${SHA256fr} MS-CEINTL.vscode-language-pack-fr-${VS_FRENCH_VERSION}.vsix" | sha256sum -c - && \
-    bsdtar -xf MS-CEINTL.vscode-language-pack-fr-$VS_FRENCH_VERSION.vsix extension && \
-    rm MS-CEINTL.vscode-language-pack-fr-$VS_FRENCH_VERSION.vsix && \
+    #Install French Language extension via building and copying https://github.com/microsoft/vscode-loc/tree/release/1.50
+    #https://github.com/Microsoft/vscode-loc/wiki/How-to-create-language-pack-extension#1-build-vsix 
+    VS_FRENCH_VERSION="1.50.2" && \
+    VS_LOCALE_REPO_VERSION="1.50" && \
```
Comment (wg102) [inline base/Dockerfile:584]: See comment above about versions

File: base/Dockerfile (hunk 1 above)
Comment (Jose-Matsuda) [reply base/Dockerfile]: My initial reasoning for using `1.50` as it's the second most up to date and I know that they would not mess with it. I do suppose that if it's under a **release** then they would not change it.

Comment (Jose-Matsuda) [reply base/Dockerfile]: Upon further review, I don't think `code-server` is compatible with 1.52 currently. The most recent version I can get to be recognized is `1.50.2
` 
image

**Pull request merged**